File: core/core/src/main/java/cn/netdiscovery/core/domain/Page.java
package cn.netdiscovery.core.domain;

import cn.netdiscovery.core.parser.selector.Html;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/**
 * Created by tony on 2017/12/23.
 */
public class Page implements Serializable {

    private static final long serialVersionUID = -3671492770100689214L;

    @Getter
    @Setter
    private String url;

    @Getter
    @Setter
    private int statusCode;//响应状态码

    @Getter
    @Setter
    private Html html;//response content

    @Getter
    private Request request;

    @Getter
    private ResultItems resultItems = new ResultItems();

    public void setRequest(Request request) {
        this.request = request;
        this.resultItems.setRequest(request);
    }

    public Page putField(String key, Object field) {
        resultItems.put(key, field);
        return this;
    }

    public Object getField(String key) {
        return resultItems.get(key);
    }

    @Override
    public boolean equals(Object o) {

        if (this == o) return true;

        if (o == null || getClass() != o.getClass()) return false;

        Page page = (Page) o;

        return url.equals(page.url);
    }

    @Override
    public int hashCode() {
        return url.hashCode();
    }
}

File: core/core/src/main/kotlin/cn/netdiscovery/core/domain/Page.kt
package cn.netdiscovery.core.domain

import cn.netdiscovery.core.parser.selector.Html
import java.io.Serializable

/**
 *
 * @FileName:
 *          cn.netdiscovery.core.domain.Page
 * @author: Tony Shen
 * @date: 2020-10-10 15:24
 * @version: V1.0 <描述当前版本功能>
 */
class Page(
    var url: String? = null,
    var statusCode:Int = 0,//响应状态码
    var html: Html? = null, //response content
    var request: Request? = null,
    val resultItems:ResultItems = ResultItems()
) : Serializable {

    fun addRequest(request: Request?) {
        this.request = request
        resultItems.request = request
    }

    fun putField(key: String?, field: Any): Page {
        resultItems.put(key!!, field)
        return this
    }

    fun getField(key: String?): Any? {
        return resultItems.get<Any>(key!!)
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val page = o as Page
        return url == page.url
    }

    override fun hashCode(): Int {
        return url.hashCode()
    }
}